app/src/main/java/com/github/h3lp3rs/h3lp/MySkillsActivity.kt

```diff
@@ -0,0 +1,84 @@
+package com.github.h3lp3rs.h3lp
+
+import android.content.Intent
+import android.os.Build
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.view.View
+import android.widget.AutoCompleteTextView
+import android.widget.EditText
+import androidx.annotation.RequiresApi
+import androidx.appcompat.app.AlertDialog
+import com.github.h3lp3rs.h3lp.dataClass.helperSkills
+import com.google.android.material.switchmaterial.SwitchMaterial
+import com.google.gson.Gson
+
+class MySkillsActivity : AppCompatActivity() {
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_my_skills)
+        loadData()
+    }
+
+    @RequiresApi(Build.VERSION_CODES.O)
+    override fun onStop() {
+        super.onStop()
+        saveData()
+    }
+
+    fun backHome( view: View){
+        val intent = Intent(this, MainPageActivity::class.java)
+        startActivity(intent)
```

Comment:
The use of startActivity will restart the activity from scratch which is not very efficient. We do not want the onCreate method to be called everytime, we should make use of the onResume method as well. I haven't looked at this in details, but this is an interesting post: https://stackoverflow.com/questions/12408719/resume-activity-in-android

Reply:
I will change that.

Thanks for the link !